unit_tests/RecordUnitTests.d

```diff
@@ -524,3 +524,25 @@ D");
 
 }
 
+@("issue #foo - translate record with name from macro")
+unittest {
+    assertTranslates(
+    q{
+
+#define __FSID_T_TYPE struct { int __val[2]; }
+typedef  __FSID_T_TYPE __fsid_t;
+typedef __fsid_t fsid_t;
+
+},
+    q{extern (C):
```

Comment:
It depends on which editor is being used. For example, in TextMate for Ruby, for heredoc strings that uses a symbol which is recognized as a language it will highlight the content of that string. Example:

```ruby
a = <<-C
int b = 3;
C

a = <<-FOO
int b = 3;
FOO
```

It plan to do the same thing with the D bundle for TextMate. That will eventually propagate to GitHub.

The approach of using `q"SYMBOL` for code is much more flexible because it can handle HTML, JSON, CSS and other code as well that is not D code.